File: SpringBootJpaTest/src/main/java/com/example/springbootjpatest/domain/OrderModel.java
package com.example.springbootjpatest.domain;


import lombok.*;
import org.hibernate.Hibernate;

import javax.persistence.*;
import java.util.Objects;


@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Data
@Table(name = "orders")
public class OrderModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "order_code", nullable = false)
    private String orderCode;

    @Column(name = "order_username", nullable = false)
    private String orderUsername;

    @Column(name = "shop_name", nullable = false)
    private String shopName;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        OrderModel that = (OrderModel) o;
        return id != null && Objects.equals(id, that.id);
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}

File: SpringKotlin-Simple-Example/src/main/kotlin/com/example/springkotlinsimpleexample/domain/OrderModel.kt
package com.example.springkotlinsimpleexample.domain

import com.example.springkotlinsimpleexample.domain.dto.CreateOrderModelDTO
import com.example.springkotlinsimpleexample.domain.dto.ReadOrderModelDTO
import org.hibernate.Hibernate
import java.time.OffsetDateTime
import java.util.Date
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity
data class OrderModel(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int? = null,
    val orderCode: String,
    val orderUserName: String,
    val orderUserPhone: String,
    val orderUserAddress: String,
    val orderUserEmail: String,
    val orderUserComment: String,
    val orderCreatedDate: OffsetDateTime = OffsetDateTime.now(),
) {
    fun getReadOrderDTO(): ReadOrderModelDTO {
        return ReadOrderModelDTO(
            id = id,
            orderCode = orderCode,
            orderUserName = orderUserName,
            orderUserPhone = orderUserPhone,
            orderUserAddress = orderUserAddress,
            orderUserEmail = orderUserEmail,
            orderUserComment = orderUserComment,
            orderCreatedDate = orderCreatedDate
        )
    }

    fun createOrderModelDTO(): CreateOrderModelDTO {
        return CreateOrderModelDTO(
            orderCode = orderCode,
            orderUserName = orderUserName,
            orderUserPhone = orderUserPhone,
            orderUserAddress = orderUserAddress,
            orderUserEmail = orderUserEmail,
            orderUserComment = orderUserComment
        )
    }


    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
        other as OrderModel

        return id != null && id == other.id
    }

    override fun hashCode(): Int = javaClass.hashCode()

    @Override
    override fun toString(): String {
        return this::class.simpleName + "(id = $id , orderCode = $orderCode , orderUserName = $orderUserName , orderUserPhone = $orderUserPhone , orderUserAddress = $orderUserAddress , orderUserEmail = $orderUserEmail , orderUserComment = $orderUserComment , orderCreatedDate = $orderCreatedDate )"
    }

}
